File: Java/minimum-number-of-vertices-to-reach-all-nodes.java
class Solution {
    public List<Integer> findSmallestSetOfVertices(int n, List<List<Integer>> edges) {
        Integer inDegree[] = new Integer[n];
        for(int i = 0; i < n; i++){
            inDegree[i] = 0;
        }
        for(List<Integer>lst: edges){
           inDegree[lst.get(1)]++;
        }
        List<Integer> ansList = new ArrayList<Integer>();
        for(int i = 0; i < n; i++){
            if(inDegree[i] == 0){
                ansList.add(i);
            }
        }
        return ansList;
    }
}

File: Graph/Topological-Sort/minimum-number-of-vertices-to-reach-all-nodes.cpp
class Solution {
public:
    vector<int> findSmallestSetOfVertices(int n, vector<vector<int>>& edges) {
        vector<int> inDegree(n, 0);
        for(auto &edge: edges){
            inDegree[edge[1]]++;
        }
        vector<int> ansVector;
        for(int i = 0; i < n; i++){
            if(inDegree[i] == 0){
                ansVector.push_back(i);
            }
        }
        return ansVector;
    }
};
